client/selfrepair/normandy_driver.js

```diff
@@ -2,27 +2,44 @@ import uuid from 'node-uuid';
 
 import Mozilla from './uitour.js';
 
+export const STORAGE_DURABILITY_KEY = '_storageDurability';
+
 /**
  * Storage class that uses window.localStorage as it's backing store.
  */
-class LocalStorage {
+export class LocalStorage {
   /**
    * @param    {string} prefix Prefix to append to all incoming keys.
    */
   constructor(prefix) {
     this.prefix = prefix;
   }
 
+  async isDurable() {
+    const durabilityStatus = localStorage.getItem(STORAGE_DURABILITY_KEY);
+    return (parseInt(durabilityStatus, 10) >= 2);
+  }
+
   _makeKey(key) {
     return `${this.prefix}-${key}`;
   }
 
   async getItem(key) {
-    return localStorage.getItem(this._makeKey(key));
+    const storageIsDurable = await this.isDurable();
+    if (!storageIsDurable && !this.testing) {
+      throw new Error('Storage durability unconfirmed');
+    }
+    const val = localStorage.getItem(this._makeKey(key));
+    try {
+      return JSON.parse(val);
```

Comment:
If you're adding this, you might want to update show-heartbeat implementation to deal with this, and the docs for the storage spec to specify that input and output is JSON-ified during storage. This will be important in the future when storing, say, date objects.